### ISSUE TITLE
[GraphQL] Can not query documents with conditional and required fields

### ISSUE THREAD

**Issue by r0skar** [payloadcms/payload#1052]
# Bug Report

Using the conditional rendering logic, I have a `story` entity that contains a `type` field. Based on the `type`, I render certain fields. eg. if `type` equals `text`, I add `headline` field and so on. The fields themselves have a  `required` attribute as well:

![image](https://user-images.githubusercontent.com/33655345/188142068-d99641b7-8db7-45e1-9991-f92c78f55dbd.png)

This works as expected in the admin dashboard, but when I query the story with graphql, it complains about the fact that some stories return null for the headline: `Cannot return null for non-nullable field`

```gql
fragment Story on Story {
  type
  # This contains the `headline` field
  text {
    header {
      headline
      tagline
    }
  }
  # An `image` story has no `headline` field and therefor throws an error
  image {
    media {
      url
    }
  }
}
```

## Expected Behavior

I can query the data without errors and keep nested required fields.

## Current Behavior

The query fails with `Cannot return null for non-nullable field`

## Additional Information

- payload version: `1.0.27`

---

Edit: Not sure about the `Bug` label anymore, because upon thinking about it, I guess it makes sense that this fails, because GQL is requesting a field that is not available, but I am wondering if this could in theory be solved on a GQL resolver level? If not, I will have to create separate queries for each type (which might not be the worst idea anyways).

**Comment by DanRibbens** [comment]
I've reproduced the issue and will have a fix ready soon!

Thanks for reporting.

**Comment by DanRibbens** [comment]
The fix was released in v1.0.32.

With the change, the graphql schema will not require sub fields of conditionals. You may still want to create your own schemas as you mentioned, but you at least now should have accurately generated schemas.
